security-council-rehearsals/step-1/README.md

```diff
@@ -0,0 +1,268 @@
+
+Now that we have verified the transaction performs the right
+operation, we need to extract the domain hash and the message hash to
+approve.
+
+Go back to the "Summary" tab, and find the first
+`GnosisSafe.domainSeparator` call. This call's return value will be
+the domain hash that will show up in your Ledger.
+
+Here is an example screenshot. Note that the hash value may be
+different:
+
+![](./images/3-tenderly-hashes.png)
+
+Right before the `GnosisSafe.domainSeparator` call, you will see a
+call to `GnosisSafe.encodeTransactionData`. Its return value will be a
+concatenation of `0x1901`, the domain hash, and the message hash (last 32 bytes/64 hex characters):
+`0x1901[domain hash][message hash]`.
+
+Here is an example screenshot. Note that the hash value may be
+different:
+
+![](./images/4-tenderly-hashes.png)
+
+Note down both the domain hash and the message hash. You will need to
+compare them with the ones displayed in your terminal AND on the Ledger screen at signing.
+
+### 4. Approve the signature on your ledger
+
+Once the validations are done, it's time to actually sign the
+transaction. Make sure your ledger is still unlocked and run the
+following:
+
+``` shell
+cd src/improvements/tasks/<network>/rehearsals/<rehearsal-task-name>
+just --dotenv-path $(pwd)/.env --justfile ../../../../single.just sign 0 # Again, you may change the integer from '0' to your own specific derivation path value.
+```
+
+> [!IMPORTANT] This is the most security critical part of the
+> playbook: make sure the domain hash and message hash in the
+> following three places match:
+
+1. In your terminal output.
+2. On your Ledger screen.
+2. In the Tenderly simulation. You should use the same Tenderly
+   simulation as the one you used to verify the state diffs, instead
+   of opening the new one printed in the console.
+
+After verification, sign the transaction. You will see the `Data`,
+`Signer` and `Signature` printed in the console. Format should be
+something like this:
+
+```
+Data:  <DATA>
+Signer: <ADDRESS>
+Signature: <SIGNATURE>
+```
+
+Double check the signer address is the right one.
+
+### 5. Send the output to Facilitator(s)
+
+Nothing has occurred onchain - these are offchain signatures which
+will be collected by Facilitators for execution. Execution can occur
+by anyone once a threshold of signatures are collected, so a
+Facilitator will do the final execution for convenience.
+
+Format should be something like this:
+
+```
+Data:  <DATA>
+Signer: <ADDRESS>
+Signature: <SIGNATURE>
+```
+
+Share the `Data`, `Signer` and `Signature` with the Facilitator, and
+congrats, you are done!
+
+## [For Facilitator ONLY] How to prepare and execute the rehearsal
+
+### [Before the rehearsal] Prepare the rehearsal
+
+#### 1. Create the Council Safe
+
+The first step of preparing the rehearsal is creating a multisig for
+the participants using the [Safe
+UI](https://app.safe.global/new-safe/create?chain=eth).
```

Comment:
Does this safe need to be in any particular network?

Reply:
pretty sure rehearsals are always performed on mainnet, I've just asked.